tmt/base.py

```diff
@@ -821,6 +826,10 @@ def __init__(self, id_=None, tree=None, context=None):
             if id_ is None:
                 raise tmt.utils.GeneralError(
                     "No last run id found. Have you executed any run?")
+        id_specified = context.params.get('last') or context.params.get('id')
+        if context.params.get('follow') and not id_specified:
```

Comment:
Specifying `--id` ends up with an error. Let's simplify the check a bit:

```python
if context.params.get('follow') and id_ is None:
```

And remove the preceding line altogether.